docs/serving/configuring-autoscaling.md

```diff
@@ -177,16 +178,47 @@ template.
 
 ```
 spec:
-  template:
-    metadata:
-      annotations:
-        autoscaling.knative.dev/metric: cpu
-        autoscaling.knative.dev/target: 70
-        autoscaling.knative.dev/class: hpa.autoscaling.knative.dev
+ template:
+  metadata:
+   annotations:
+    autoscaling.knative.dev/metric: cpu
+    autoscaling.knative.dev/target: 70
+    autoscaling.knative.dev/class: hpa.autoscaling.knative.dev
 ```
+## Using the recommended autoscaling reconciler for custom Go implementations
```

Comment:
I found that this is the first place in the Serving documentation where we mention a "reconciler" and unfortunately that object is not defined anywhere?

Can we explain what this is and what it does? At this point, it sounds like its only required for autoscaling in serving? Is this the only place its needed (its otherwise just mentioned in [Eventing's "create a source" topic](https://knative.dev/docs/eventing/samples/writing-a-source/03-reconcile-sources/) but not explained there either).

cc/ @markusthoemmes

Reply:
It's also known as the controller as far as I know:
https://kubernetes.io/docs/concepts/architecture/controller/

Markus said we could just link to these docs. It's on the autoscaling roadmap to provide a sample but it's not high priority right now. I can rename this in the docs from reconciler -> controller if it makes sense to do that.

Reply:
@abrennan89: Sounds good to me. I found this too which might help https://godoc.org/k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler

Reply:
I think maybe it's called reconciler in this section because the actual one used is called `autoscaling-base-reconciler` rather than controller. I'm making some updates anyway in #1980 that should make it clearer, you can take a look and lmk though.